manifest-go117.tmpl

```diff
@@ -0,0 +1,20 @@
+image: rancher/golang:1.17-windowsservercore
+manifests:
+  -
+    image: rancher/golang:1.17-windowsservercore-1809
+    platform:
+      architecture: amd64
+      os: windows
+      version: 1809
+  -
+    image: rancher/golang:1.17-windowsservercore-20H2
```

Comment:
We only need to add 20h2 images for 1.17 as we are going to use the upstream golang images for 1809 and 2022. We don't need any of the linux images for 1.17 as they are all available upstream and we have not been publishing them under rancher/golang for some time.

The only new files should be for 20h2 and the manifest template for 1.17 should only include 20H2 as well. Same for the drone pipelines, only `name: windows-go117-20H2` is needed.

This gives us the easiest approach to archive rancher/golang once 20H2 support has ended in a few months as we would already be using the upstream 1809 and 2022 images.